src/esp/assets/ResourceManager.cpp

```diff
@@ -1293,18 +1295,26 @@ bool ResourceManager::loadRenderAssetIMesh(const AssetInfo& info) {
     // only one mesh, treat as before
     meshData = fileImporter_->mesh(0);
   } else {
-    // build list of meshDatas from importer
-    std::vector<Mn::Trade::MeshData> meshVec;
-    meshVec.reserve(meshCount);
+    Cr::Containers::Optional<Mn::Trade::SceneData> scene =
+        fileImporter_->scene(fileImporter_->defaultScene());
+    // std::vector<Mn::Trade::MeshData> meshVec;
+    // meshVec.reserve(meshCount);
+    Cr::Containers::Array<Mn::Trade::MeshData> flattenedMeshes;
+    for (const Cr::Containers::Triple<Mn::UnsignedInt, Mn::Int, Mn::Matrix4>&
+             meshTransformation :
+         Mn::SceneTools::flattenMeshHierarchy3D(*scene)) {
+      if (Cr::Containers::Optional<Mn::Trade::MeshData> mesh =
+              fileImporter_->mesh(meshTransformation.first())) {
+        arrayAppend(flattenedMeshes, Mn::MeshTools::transform3D(
+                                         *mesh, meshTransformation.third()));
+      }
+    }
+
     // build view
     std::vector<Cr::Containers::Reference<const Mn::Trade::MeshData>> meshView;
```

Comment:
Does this still need to be a vector? Shouldn't it be an std::array or something now?

Reply:
Yep, forgot I wanted to use a corrade array instead.